ViewModel/HyvaGrid/Navigation.php

```diff
@@ -207,21 +207,24 @@ private function getQueryParam(string $param)
 
     public function getSearchCriteria(): SearchCriteriaInterface
     {
-        if ($this->isPagerEnabled()) {
-            $this->searchCriteriaBuilder->setPageSize($this->getPageSize());
-            // The requested page number has to be used here instead of the current page number,
-            // because the current page number requires the search criteria to load the records from the source,
-            // which creates a circular dependency. This means the grid source has to deal with the case
-            // when the page number on the search criteria is larger than the available pages.
-            // However, all page links returned by this class will never go beyond the last page.
-            $this->searchCriteriaBuilder->setCurrentPage($this->getRequestedPageNumber());
+        if (!$this->memorizedSearchCriteria) {
+            if ($this->isPagerEnabled()) {
+                $this->searchCriteriaBuilder->setPageSize($this->getPageSize());
+                // The requested page number has to be used here instead of the current page number,
+                // because the current page number requires the search criteria to load the records from the source,
+                // which creates a circular dependency. This means the grid source has to deal with the case
+                // when the page number on the search criteria is larger than the available pages.
+                // However, all page links returned by this class will never go beyond the last page.
+                $this->searchCriteriaBuilder->setCurrentPage($this->getRequestedPageNumber());
+            }
+            $this->searchCriteriaBuilder->addSortOrder($this->createSortOrder());
+            map(function (GridFilterInterface $filter): void {
+                $filter->apply($this->searchCriteriaBuilder);
+            }, filter(map([$this, 'getFilter'], keys($this->columnDefinitions))));
+
+            $this->memorizedSearchCriteria = $this->searchCriteriaBuilder->create();
         }
-        $this->searchCriteriaBuilder->addSortOrder($this->createSortOrder());
-        map(function (GridFilterInterface $filter): void {
-            $filter->apply($this->searchCriteriaBuilder);
-        }, filter(map([$this, 'getFilter'], keys($this->columnDefinitions))));
-
-        return $this->searchCriteriaBuilder->create();
+        return $this->memorizedSearchCriteria;
```

Comment:
Have you profiled if memoizing the search criteria gives any benefits?
From what I can see it shouldn't make a noticable difference.

Reply:
In the new context of the export the difference is partly because of  performance. Because every time i call the getSearchCriteria-Method the function instances a new searchCriteria with applying all the filters. So It is not possible to iterate through a data source, when not passing a search criteria to the getRows method